plugins/connection/network_cli.py

```diff
@@ -1370,3 +1413,123 @@ def _needs_cache_invalidation(self, command):
         if (self._is_in_config_mode()) or (to_text(command) in cfg_cmds):
             invalidate = True
         return invalidate
+
+    def _send_playback(self, command):
+        """Send the fixture response rather than the actual command."""
+
+        test_parameters = self.get_option("test_parameters")
+
+        self._send_sequence += 1
+
+        fixture_file = os.path.join(
+            test_parameters["fixture_directory"],
+            "%s.json" % str(self._send_sequence).zfill(5),
+        )
+
+        if not os.path.exists(fixture_file):
+            raise AnsibleError(
+                "Fixture file %s does not exist." % fixture_file
+            )
+
+        with open(fixture_file, "r") as f:
+            fixture = json.load(f)
+
+        if fixture["command"] != command.decode("utf-8"):
+            raise AssertionError(
+                "Fixture command %s does not match command %s."
+                % (fixture["command"], command)
+            )
+        if fixture["response_type"] == "json":
+            return json.dumps(fixture["response"])
+        return "\n".join(fixture["response"])
+
+    def _send_post(self, command, response):
+        """Proxy the response to the send() method"""
+        test_parameters = self.get_option("test_parameters")
+        if not test_parameters:
+            return response
+
+        # Don't record the terminal commands
+        if command.decode("utf-8").startswith("terminal"):
+            return response
+
+        self._send_sequence += 1
+
+        fixture_file = os.path.join(
+            test_parameters["fixture_directory"],
+            "%s.json" % str(self._send_sequence).zfill(5),
```

Comment:
Let string formatting do all the hard work?
```suggestion
            "%05d.json" % self._send_sequence,
```